db/migrate/20131114084718_extend_user_role.rb

```diff
@@ -22,4 +22,15 @@ def down
     remove_column :user_roles, :owner_type
     add_foreign_key 'user_roles', 'users', :name => 'user_roles_user_id_fk'
   end
+
+  private
+
+  # Rails 5.2 compatible check
+  def compatible_foreign_key_exists?(source, target = nil, **options)
+    if Rails.version.start_with?('5.2')
+      foreign_key_exists?(source, options)
+    else
+      foreign_key_exists?(source, target, options)
+    end
+  end
```

Comment:
Should this be monkey patched into  `ActiveRecord::ConnectionAdapters::SchemaStatements` ? Perhaps overriding the `foreign_key_exists?` method so that all plugins get it for free with some warning and once we upgrade all we need to do is drop the monkey patch?

Reply:
:+1: yeah, definitely. I wasn't thinking about plugins.

Reply:
I've put it to separate initializer, so it should be fairly easy to remove once the support for `Rails 5.2` is dropped. So I would say there's no need for warning?
Plugins will still fail unfortunatelly, so they'll need to do the same fix to work on Rails 6, but this will ensure the compatibility with Rails 5.2 for them.